packages/core/saved-objects/core-saved-objects-api-server-internal/src/lib/apis/find.ts

```diff
@@ -266,13 +268,24 @@ export const performFind = async <T = unknown, A = unknown>(
   const migratedDocuments: Array<SavedObjectsFindResult<T>> = [];
   try {
     for (const savedObject of savedObjects) {
+      let migrated: SavedObjectUnsanitizedDoc | undefined;
       const { sort, score, ...rawObject } = savedObject;
-      const migrated = disableExtensions
-        ? migrationHelper.migrateStorageDocument(rawObject)
-        : await migrationHelper.migrateAndDecryptStorageDocument({
-            document: rawObject,
-            typeMap: redactTypeMap,
-          });
+
+      if (fields === undefined || fields.length === 0) {
```

Comment:
I _think_ this logic is backwards. IIUC we want to avoid migration if a subset of the fields are requested, otherwise we are getting all fields and we can migrate the doc? Should we also check if the fields array contains all fields, or is that overkill?

```suggestion
      if (fields !== undefined && fields.length > 0) {
```

Reply:
specifying `fields: []` is equivalent to saying "don't give me any fields but the root fields" so I changed it to also consider an empty array a case where we shouldn't migrate